### DISCUSSION TITLE
`ShouldBeUnique` does not always prevent duplicate jobs

### DISCUSSION THREAD

**Discussion by liamkeily** [laravel/framework#50219] [General]
I am using `ShouldBeUnique` on Jobs in a project and noticed that duplicate jobs are still being queued. On investigation it appears that `ShouldBeUnique` only works if you dispatch a job in certain ways. 

I have put together a test to prove the various behaviours of `ShouldBeUnique` https://github.com/liamkeily/laravel-queue-behaviour/commit/ff34c3e5950650fe901cddd02eeddc1dd1f43d66#diff-88de2e685736b4d69dd9ac122767ca9580e3feb74cc821bc1f109accb2ea9017

To summarise:

1.  The `dispatch` helper works correctly and prevents duplicate jobs
2. `Bus::dispatch` ignores it and allows duplicate jobs
3. `Bus::batch`  ignores it and allows duplicate jobs
4. `Bus::chain` removes duplicates in an individual chain, but allows another chain to add duplicates

I'm not sure if this is considered a bug or expected behaviour, but it seems confusing so opening as a discussion for thoughts.

**Comment by chrisrickard** [comment]
Yup, same issue... turns out [it's expected behaviour](https://laravel.com/docs/10.x/queues#unique-jobs)

![image](https://github.com/user-attachments/assets/7b937d15-f728-4f05-a782-531939c4f171)

**Reply by bogutskyy** [reply]
`Bus::dispatch` is not about batches.
